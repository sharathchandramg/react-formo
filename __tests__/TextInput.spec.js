import React from "react";
import { shallow, mount } from 'enzyme';
import FormO from "../src/index";
import TextInputField from "../src/fields/textInput/index";

const testSchema = {
  "type": "text",
  "name": "data_source",
  "label": "Data Source",
  "required": true,
  "editable": true,
  "hidden": false
};

const fields= require("../schema/00-basic_types.json");
// const fields1=require("../example/sampleform/src/schema/01-phone-and-email.json");

describe(FormO,()=>{
  it('Renders FormO',()=>{
    const form = mount(<FormO fields={fields}/>);
    expect(FormO).toMatchSnapshot();
  })
});

describe(TextInputField,()=>{
  it('Renders text Input field',()=>{
    const inputField=shallow(<TextInputField attributes={testSchema} />);
    expect(TextInputField).toMatchSnapshot();
  })
})

