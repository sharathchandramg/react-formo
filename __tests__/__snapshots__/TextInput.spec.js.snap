// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormO Renders FormO 1`] = `[Function]`;

exports[`FormO Renders FormO with phone-and-email 1`] = `[Function]`;

exports[`FormO Renders FormO with schema 00-basic_types 1`] = `[Function]`;

exports[`TextInputField Renders text Input field 1`] = `[Function]`;
